diff --git a/BUILD.gn b/BUILD.gn
index efaf09033..7c65c83ca 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -11,7 +11,9 @@ if (angle_use_wayland) {
   import("//build_overrides/wayland.gni")
 }
 if (angle_has_build) {
-  import("//build/config/linux/pkg_config.gni")
+  if (!is_ohos) {
+    import("//build/config/linux/pkg_config.gni")
+  }
   import("//build/config/sanitizers/sanitizers.gni")
   import("//build/config/ui.gni")
   import("//testing/libfuzzer/fuzzer_test.gni")
@@ -83,7 +85,7 @@ if (angle_build_all) {
       ":translator_fuzzer",
       ":xxhash_fuzzer",
       "$angle_root/samples:angle_samples",
-      "$angle_root/src/tests:angle_tests",
+      #"$angle_root/src/tests:angle_tests",
     ]
     if (angle_enable_cl) {
       deps += [ "$angle_root/src/libOpenCL:angle_cl" ]
@@ -141,7 +143,7 @@ config("internal_config") {
     if (angle_is_winuwp) {
       defines += [ "ANGLE_IS_WINUWP" ]
     }
-  } else if (is_linux || is_chromeos) {
+  } else if (is_linux || is_chromeos || is_ohos) {
     defines += [ "ANGLE_IS_LINUX" ]
   }
 
@@ -443,7 +445,7 @@ angle_static_library("angle_common") {
     all_dependent_configs = [ ":angle_disable_pool_alloc" ]
   }
 
-  if (is_linux || is_chromeos) {
+  if (is_linux || is_chromeos || is_ohos) {
     libs = [ "dl" ]
   }
 
@@ -486,7 +488,8 @@ angle_static_library("angle_image_util") {
   sources = libangle_image_util_sources
   public_configs += [ ":angle_image_util_config" ]
   public_deps = [ ":angle_image_util_headers" ]
-
+  
+  angle_has_astc_encoder = false
   if (angle_has_astc_encoder) {
     public_deps += [ "third_party/astc-encoder:astcenc" ]
     include_dirs = [ "third_party/astc-encoder/src/Source/" ]
@@ -561,7 +564,7 @@ angle_static_library("angle_gpu_info_util") {
     libs += [ "dxgi.lib" ]
   }
 
-  if (is_linux || is_chromeos) {
+  if (is_linux || is_chromeos || is_ohos) {
     sources += libangle_gpu_info_util_linux_sources
 
     if (angle_use_x11 && angle_has_build) {
diff --git a/gni/angle.gni b/gni/angle.gni
index b8086660a..d82d056d6 100644
--- a/gni/angle.gni
+++ b/gni/angle.gni
@@ -60,8 +60,7 @@ if (angle_has_build) {
 
   declare_args() {
     angle_use_gbm = ozone_platform_gbm
-    angle_use_x11 =
-        ozone_platform_x11 && !is_ggp && (is_linux || is_chromeos) && !is_castos
+    angle_use_x11 = ozone_platform_x11 && !is_ggp && (is_linux || is_chromeos) && !is_castos
     angle_use_wayland =
         ozone_platform_wayland && !is_ggp && is_linux && !is_castos
     angle_use_vulkan_display = (is_linux || is_chromeos) && !is_ggp
@@ -191,7 +190,9 @@ declare_args() {
 
 declare_args() {
   # ASTC emulation is only built on standalone non-android builds
-  angle_has_astc_encoder = angle_has_build && angle_standalone && !is_android
+  #angle_has_astc_encoder = angle_has_build && angle_standalone && !is_android
+  #for ohos
+  angle_has_astc_encoder = false
 }
 
 declare_args() {
@@ -255,13 +256,13 @@ declare_args() {
   angle_wayland_dir = "$angle_root/third_party/wayland"
 
   angle_vulkan_headers_dir =
-      "$angle_root/third_party/vulkan-deps/vulkan-headers/src"
+      "//third_party/vulkan-deps/vulkan-headers/src"
   angle_vulkan_loader_dir =
-      "$angle_root/third_party/vulkan-deps/vulkan-loader/src"
+      "//third_party/vulkan-deps/vulkan-loader/src"
   angle_vulkan_tools_dir =
-      "$angle_root/third_party/vulkan-deps/vulkan-tools/src"
+      "//third_party/vulkan-deps/vulkan-tools/src"
   angle_vulkan_validation_layers_dir =
-      "$angle_root/third_party/vulkan-deps/vulkan-validation-layers/src"
+      "//third_party/vulkan-deps/vulkan-validation-layers/src"
 
   angle_build_vulkan_system_info = angle_has_build && !angle_is_winuwp
 
diff --git a/include/EGL/eglplatform.h b/include/EGL/eglplatform.h
index 9ebaf00a9..fe111d115 100644
--- a/include/EGL/eglplatform.h
+++ b/include/EGL/eglplatform.h
@@ -117,13 +117,19 @@ typedef khronos_uintptr_t EGLNativeWindowType;
 
 #elif defined(__unix__) || defined(USE_X11)
 
-/* X11 (tentative)  */
-#include <X11/Xlib.h>
-#include <X11/Xutil.h>
+struct NativeWindow;
 
-typedef Display *EGLNativeDisplayType;
-typedef Pixmap   EGLNativePixmapType;
-typedef Window   EGLNativeWindowType;
+typedef void*                   EGLNativeDisplayType;
+typedef void*                   EGLNativePixmapType;
+typedef struct NativeWindow*    EGLNativeWindowType;
+
+// /* X11 (tentative)  */
+// #include <X11/Xlib.h>
+// #include <X11/Xutil.h>
+
+// typedef Display *EGLNativeDisplayType;
+// typedef Pixmap   EGLNativePixmapType;
+// typedef Window   EGLNativeWindowType;
 
 #elif defined(__APPLE__)
 
diff --git a/src/common/angle_version.h b/src/common/angle_version.h
index d9d7e8929..a97e3844b 100644
--- a/src/common/angle_version.h
+++ b/src/common/angle_version.h
@@ -14,8 +14,12 @@
 #define ANGLE_MINOR_VERSION 1
 
 #ifndef ANGLE_REVISION
+#ifndef ANGLE_COMMIT_POSITION
+#    define ANGLE_REVISION 0
+#else
 #    define ANGLE_REVISION ANGLE_COMMIT_POSITION
 #endif
+#endif
 
 #define ANGLE_STRINGIFY(x) #x
 #define ANGLE_MACRO_STRINGIFY(x) ANGLE_STRINGIFY(x)
diff --git a/src/compiler/translator/BuildSPIRV.h b/src/compiler/translator/BuildSPIRV.h
index d67bd812b..00d22d025 100644
--- a/src/compiler/translator/BuildSPIRV.h
+++ b/src/compiler/translator/BuildSPIRV.h
@@ -110,6 +110,8 @@ struct SpirvType
     SpirvTypeSpec typeSpec;
 };
 
+bool operator==(const SpirvType &a, const SpirvType &b);
+
 struct SpirvIdAndIdList
 {
     spirv::IdRef id;
diff --git a/src/libANGLE/renderer/gl/BUILD.gn b/src/libANGLE/renderer/gl/BUILD.gn
index c675c2607..6cba114ef 100644
--- a/src/libANGLE/renderer/gl/BUILD.gn
+++ b/src/libANGLE/renderer/gl/BUILD.gn
@@ -125,7 +125,7 @@ if (angle_use_x11) {
   ]
 }
 
-if (is_android || is_linux || is_chromeos) {
+if (is_android || is_linux || is_chromeos || is_ohos) {
   _gl_backend_sources += [
     "egl/ContextEGL.cpp",
     "egl/ContextEGL.h",
@@ -264,7 +264,7 @@ angle_source_set("angle_gl_backend") {
       "Xext",
     ]
   }
-  if (is_android || is_linux || is_chromeos) {
+  if (is_android || is_linux || is_chromeos || is_ohos) {
     deps += [ "$angle_root/src/common/linux:angle_dma_buf" ]
   }
   if (is_apple) {
diff --git a/src/libGLESv2.gni b/src/libGLESv2.gni
index 723d7f99f..dab8b887c 100644
--- a/src/libGLESv2.gni
+++ b/src/libGLESv2.gni
@@ -87,7 +87,7 @@ xxhash_sources = [
   "src/common/third_party/xxhash/xxhash.h",
 ]
 
-if (is_linux || is_chromeos || is_android || is_fuchsia) {
+if (is_linux || is_chromeos || is_android || is_fuchsia || is_ohos) {
   libangle_common_sources += [
     "src/common/system_utils_linux.cpp",
     "src/common/system_utils_posix.cpp",
